### PR TITLE
Add code intelligence and CI checks for GQL queries

### PR DESCRIPTION
**Pull request by eseliger**
Implements code intelligence on hover for gql queries in TypeScript and adds a CI check that validates that no invalid fields are used in the strings.

Depends on:
https://github.com/Quramy/ts-graphql-plugin/pull/107

Errors: 
image

Code intel:
image

Deprecations:
image

At a later stage, we can also take a look at the typegen it features, but for now wanted to get that in, which is an essential part to making integration tests safely replace e2e tests.

### CODE REVIEW

File: shared/src/e2e/tsconfig.json [hunk 1]
```diff
@@ -4,6 +4,13 @@
   "compilerOptions": {
     "types": ["mocha", "node"],
     "module": "commonjs",
+    "plugins": [
```
Comment (felixfbecker) [inline shared/src/e2e/tsconfig.json:7]: This can't be in the base config?

Comment (eseliger) [reply shared/src/e2e/tsconfig.json:7]: Sadly the schema path is resolved on the project root, which is where the tsconfig lives, so I didn't find a way

File: web/src/search/backend.tsx [hunk 2]
```diff
@@ -9,28 +9,28 @@ import { mutateGraphQL, queryGraphQL } from '../backend/graphql'
...
-          text
-          html
-      }
-      highlights {
-          line
-          character
-          length
-      }
-  }
+const genericSearchResultInterfaceFields = `
```
[the remark this answers is not in the document]
Comment (eseliger) [reply web/src/search/backend.tsx:12]: Huh, you are right! Didn't know this is an actual gql type, I thought it is just properties they have in common

File: web/src/marketing/backend.tsx [hunk 3]
```diff
@@ -104,7 +104,7 @@ export type SurveyResponseConnectionAggregates = Exclude<GQL.ISurveyResponseConn
 export function fetchSurveyResponseAggregates(): Observable<SurveyResponseConnectionAggregates> {
     return queryGraphQL(
         gql`
-            query FetchSurveyResponseAggregates() {
+            query FetchSurveyResponseAggregates {
```
Comment (felixfbecker) [inline web/src/marketing/backend.tsx:107]: It always weirded me out that our GraphQL backend doesn't reject this (like the spec dictates!).

File: web/src/search/backend.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (eseliger) [reply web/src/search/backend.tsx:13]: Broken in `graphql-go`, added a TODO comment

**Pull request merged**